client/asset/eth/eth.go

```diff
@@ -408,17 +409,62 @@ type TokenWallet struct {
 	token    *dexeth.Token
 }
 
+const maxProportionOfBlockGasLimitToUse = 4
+
 // Info returns basic information about the wallet and asset.
 func (*ETHWallet) Info() *asset.WalletInfo {
+	gasCeil := ethconfig.Defaults.Miner.GasCeil
+	var maxSwapGas, maxRedeemGas uint64
+
+	for _, gases := range dexeth.VersionedGases {
+		if gases.Swap > maxSwapGas {
+			maxSwapGas = gases.Swap
+		}
+		if gases.Redeem > maxRedeemGas {
+			maxRedeemGas = gases.Redeem
+		}
+	}
+
+	WalletInfo.MaxSwapsInTx = int(gasCeil / maxProportionOfBlockGasLimitToUse / maxSwapGas)
+	WalletInfo.MaxRedeemsInTx = int(gasCeil / maxProportionOfBlockGasLimitToUse / maxRedeemGas)
+
```

Comment:
It looks like all of these numbers are constant and we don't need to calculate it every time. Could also use the swapAdd or redeemAdd to be more precise.

I guess the max number would be something like 30000000/4/135000 or 55 swaps for eth?

Reply:
I wanted to do a calculation so that this doesn't need to be updated if a new version of the ETH contract is added. I guess I could do the calculation in `init` to avoid doing it on each call to `Info`.

> I guess the max number would be something like 30000000/4/135000 or 55 swaps for eth?

Yeah, you can see the exact values in `eth_test.go`.

Reply:
I've updated it to be calculated just once when the wallet is created. I wanted to do a calculation instead of hardcoding the value to avoid having to update this when new versions of the contract are created.

> I guess the max number would be something like 30000000/4/135000 or 55 swaps for eth?
Yes its 55 swaps and 119 redeems for eth.